### PR TITLE
homework 3

### PR DESCRIPTION
**Pull request by SergeyKostikov**

### CODE REVIEW

File: js-core/homeworks/homework-3/src/main.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    {
+      argument1: param1,
+      argument2: param2,
+      argument3: param3,
+    }
+*/
+
+function simpleObjectGenerator(one, two, three) {
+  const obj = {
+    arg1:one,
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-3/src/main.js]: It's not really argument1 property :)

File: js-core/homeworks/homework-3/src/main.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    result = num;
+  }else if(num%3==0 && num%5!=0){
+    result = `Fizz`;
+  }else if(num%3!=0 && num%5==0){
+    result = 'Buzz';
+  }else if(num%3==0 && num%5==0){
+    result = 'FizzBuzz';
+  }
+
+  console.log(result);
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-3/src/main.js]: Probably we need to return a value not log it

File: js-core/homeworks/homework-3/src/main.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+-> @@ SUPER !
+  Напишите функцию, которая будет
+   возвращать 'Fizz' если передаваемый параметр кратен 3,
+   'Buzz', если передаваемый параметр кратен 5,
+   'FizzBuzz' - если параметер кратен 3 и 5
+   Если передаваемое число не кратно 3 или 5, то вернуть указанный параметр
+*/
+
+function fizzBuzz(num) {
+
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-3/src/main.js]: it's fine, but can be simplified

Comment (OlegLustenko) [reply js-core/homeworks/homework-3/src/main.js]: You can try to think about a solution in which you concatenate a string

**Comment by SergeyKostikov** [comment]
change the  FizzBuzz function and rename arguments and property for second task)))

File: js-core/homeworks/homework-3/src/main.js [hunk 4]
```diff
@@ -128,21 +132,23 @@ console.log(myName);
...
-  }else if(num%3==0 && num%5==0){
-    result = 'FizzBuzz';
+    strFizzBuzz = num;
+  }else{
+    strFizzBuzz = `${fizz}${buzz}`;
   }
-
-  console.log(result);
-  
+  return strFizzBuzz;
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-3/src/main.js]: cool, but there still some optimizations left :)

I leave it as-is and going to merge it in friday

**Review by OlegLustenko — DISMISSED** [review]

**Review by OlegLustenko — APPROVED** [review]

**Pull request merged**